lambdas/services/document_upload_review_service.py

```diff
@@ -0,0 +1,120 @@
+    def table_name(self) -> str:
+        return os.environ.get("DOCUMENT_REVIEW_DYNAMODB_NAME")
+
+    @property
+    def model_class(self) -> type:
+        return DocumentUploadReviewReference
+
+    @property
+    def s3_bucket(self) -> str:
+        return os.environ.get("DOCUMENT_REVIEW_S3_BUCKET_NAME")
+
+    def query_docs_pending_review_by_custodian(
+        self,
+        ods_code: str,
+        limit: int | None = None,
+        start_key: dict | None = None,
+        nhs_number: str | None = None,
+        uploader: str | None = None,
+    ) -> tuple[list[DocumentUploadReviewReference], dict | None]:
+        logger.info(f"Getting review document references for custodian: {ods_code}")
+
+        filter_expression = self.build_review_query_filter(
+            nhs_number=nhs_number, uploader=uploader
+        )
+
+        if not limit:
+            limit = 50
+
+        try:
+            response = self.dynamo_service.query_table_single(
+                table_name=self.table_name,
+                search_key="Custodian",
+                search_condition=ods_code,
+                index_name="CustodianIndex",
+                limit=limit,
+                start_key=start_key,
+                filter_expression=filter_expression,
+            )
+
+            references = self._validate_review_references(response["Items"])
+
+            last_evaluated_key = response.get("LastEvaluatedKey", None)
+
+            return references, last_evaluated_key
+
+        except ClientError as e:
+            logger.error(e)
+            raise DocumentReviewException(500, LambdaError.DocumentReviewDB)
+
+    def _validate_review_references(
+        self, items: list[dict]
+    ) -> list[DocumentUploadReviewReference]:
+        try:
+            logger.info("Validating document review search response")
+            review_references = [
+                self.model_class.model_validate(item) for item in items
+            ]
+            return review_references
+        except ValidationError as e:
+            logger.error(e)
+            raise DocumentReviewException(500, LambdaError.DocumentReviewValidation)
+
+    def update_document_review_custodian(
+        self,
+        patient_documents: list[DocumentUploadReviewReference],
+        updated_ods_code: str,
+    ) -> None:
+        review_update_field = {"custodian"}
+        if not patient_documents:
+            return
+
+        for review in patient_documents:
+            logger.info("Updating document review custodian...")
+
+            if review.custodian != updated_ods_code:
+                review.custodian = updated_ods_code
+
+                self.update_document(
+                    document=review,
+                    update_fields_name=review_update_field,
+                )
+
+    def build_review_query_filter(
+        self, nhs_number: str | None = None, uploader: str | None = None
+    ) -> Attr | ConditionBase:
+        filter_builder = DynamoQueryFilterBuilder()
```

Comment:
I think you need to set `filter_builder.set_combination_operator(ConditionOperator.OR)` here
so we search NHS number or ODS code.

The user can enter an NHS number or ods code into the search filter box.